pyro/poutine/messenger.py

```diff
@@ -133,15 +134,15 @@ def _process_message(self, msg):
         Process the message by calling appropriate method of itself based
         on message type. The message is updated in place.
         """
-        method_name = "_pyro_{}".format(msg["type"])
-        if hasattr(self, method_name):
-            return getattr(self, method_name)(msg)
+        method = getattr(self, "_pyro_{}".format(msg["type"]), None)
+        if method is not None:
+            return method(msg)
```

Comment:
This and the below change are micro-optimizations included in this PR to prevent any performance regression.